packages/lib/har/harUtils.ts

```diff
@@ -64,12 +64,24 @@ export const toHarContentBase64 = (body: Buffer, mimeType?: string): HarFormatCo
   encoding: 'base64',
 });
 
-export const toHarContent = (body: string | Buffer | null, mimeType?: string): HarFormatContent => {
+export const toHarContent = (
+  saveAsString: boolean,
+  body: string | Buffer | null,
+  mimeType?: string,
+): HarFormatContent => {
   if (Buffer.isBuffer(body)) {
     if (isBinary(mimeType ? `file.${extension(mimeType)}` : null, body)) {
       return toHarContentBase64(body, mimeType);
     }
 
+    if (mimeType === 'application/json' && !saveAsString) {
+      return {
+        mimeType: mimeType ?? '',
+        size: body?.length ?? 0,
+        text: JSON.parse(body.toString('binary')),
```

Comment:
Instead of using the `text` field, I think we should rather use a new `json` field.